packages/xod-project/src/utils.js

```diff
@@ -194,6 +194,14 @@ export const isValidCharLiteral = def(
   R.both(isLikeCharLiteral, R.complement(isAmong(unescapedCharLiterals)))
 );
 
+export const isValidErrcodeLiteral = def(
+  'isValidErrcodeLiteral :: String -> Boolean',
+  R.both(
+    R.test(/^E\d{1,3}$/g),
+    R.pipe(R.tail, n => parseInt(n, 10), n => n >= 0 && n <= 127)
```

Comment:
Is `E010` a valid error literal? If so, we have a problem during conversion to C++. It will strip leading E and the value will be `010` which is octal `9` in C++ (leading zero denotes octal) which can be a big source of confusion.

Maybe it’s a good idea to forbid zero after `E` except the `E0` case